src/main/java/com/google/sps/utils/data/DataCollection.java

```diff
@@ -23,240 +28,525 @@
-   * @param trendImg html for the species image in the table
-   */
-  private static String scrapeImageLink(Element image) {
-    String imageLink = "";
-    if (image != null) {
-      imageLink = image.absUrl("src");
+
+    /**
+    * Takes in the population information from wikipedia and cleans it.
+    * 
+    * Input comes in two main formats (excluding any commas or white space characters):
+    *    1. a[b] where a is the population count and b is the reference number
+    *    2. a-b[c] where a is the lower bound population, b is the upper bound, and c is the reference number
+    * @param populationString string that needs to be cleaned
+    */
+    private static String scrapePopulation(String populationString) {
+        String pop = removeBrackets(populationString);
+        pop = pop.replaceAll("[^0-9.–-]", ""); // remove any characters other than numbers and '-'
+        return pop;
+    }
+
+    /**
+    * Takes in the trendImg element from wikipedia and extracts the trend
+    * 
+    * If the species has trend information, the format is:
+    *    <img alt="TREND" src="image" title="TREND">
+    * @param trendImg html for the trend box in the table
+    */
+    private static PopulationTrend scrapeTrend(Element trendImg) {
+        String trend;
+        if (trendImg != null) {
+            trend = trendImg.attr("alt");
+        } else {
+            return PopulationTrend.UNKNOWN;
+        }
+        return convertToPopulationTrendEnum(trend);
+    }
+
+    /**
+    * Takes in content of the scientific name cell from wikipedia and extracts the binomial name.
+    *
+    * Example: Balaena mysticetus Linnaeus, 1758 -> Balaena mysticetus
+    * @param scientificName content of the scientific name cell
+    */
+    private static String scrapeBinomial(String scientificName) {
+        String[] nameArray = scientificName.split(" ");
+        String result = nameArray[0] + " " + nameArray[1];
+        // System.out.println("before: " + scientificName + " ===== after : " + result);
+        return result;
+    }
+
+    /**
+    * Takes in the image element from wikipedia and extracts the url
+    * 
+    * If the species has an image, the format is:
+    *    <img src="IMAGE_URL" srcset="IMAGE_URL_SIZE:1.5x, IMAGE_URL_SIZE:2x">
+    * @param trendImg html for the species image in the table
+    */
+    private static String scrapeImageLink(Element image) {
+        if (image == null) {
+            return null;
+        }
+        String src = image.attr("src");
+        String srcset = image.attr("srcset");
+
+        // remove size descriptions and only keep image urls
+        ArrayList<String> images = new ArrayList<>(Arrays.asList(srcset.split(" ")));
+        images.add(0, src);
+        Iterator<String> itr = images.iterator(); 
+        while (itr.hasNext()) {
+            String url = itr.next();
+            if (url.charAt(0) != '/') {
+                itr.remove();
+            }
```

Comment:
I can update the comment on this function to be a bit more clear! The structure for an image is 
```<img src="//upload.wikimedia.org/wikipedia/commons/thumb/8/83/Canis_rufus_FWS_cropped.jpg/175px-Canis_rufus_FWS_cropped.jpg" srcset="//upload.wikimedia.org/wikipedia/commons/thumb/8/83/Canis_rufus_FWS_cropped.jpg/263px-Canis_rufus_FWS_cropped.jpg 1.5x, //upload.wikimedia.org/wikipedia/commons/thumb/8/83/Canis_rufus_FWS_cropped.jpg/350px-Canis_rufus_FWS_cropped.jpg 2x">``` 

``srcset`` is a string with all the urls beginning with a "//" and then the size descriptions separated by a space. I split srcset by spaces and just kept the urls and converted them to absolute urls

Reply:
My main interest was in understanding the format that was being parsed and what the input data looks like. I was really looking more for a comment update rather than code changes.

The main thing I couldn't figure out was what a concrete-ish example would look like. So that was why I asked about "http:" and the "1.5x", which is not a part of the URL but some additional tag that needed to be removed.